### PR TITLE
Issue 24: Update tests to check all keys for SA_STATUS_OPERATION_NOT_SUPPORTED

### PR DESCRIPTION
**Pull request by ericeberry**

### CODE REVIEW

File: reference/src/client/src/sa_engine_pkey.c [hunk 1]
```diff
@@ -936,6 +936,11 @@ static int pkey_decrypt(
     sa_crypto_cipher_context cipher_context;
     sa_status status = sa_crypto_cipher_init(&cipher_context, cipher_algorithm, SA_CIPHER_MODE_DECRYPT,
             data->private_key, parameters);
+    if (status == SA_STATUS_OPERATION_NOT_SUPPORTED) {
+        ERROR("sa_crypto_cipher_init operation not supported");
+        return -2;
```
Comment (mhabrat) [inline reference/src/client/src/sa_engine_pkey.c:941]: Why are we not using some defined enum for the return for code readability?

File: reference/src/client/test/sa_engine_pkey_decrypt.cpp [hunk 2]
```diff
@@ -91,6 +91,9 @@ TEST_P(SaEnginePkeyEncryptTest, encryptTest) {
     std::shared_ptr<EVP_MD_CTX> evp_md_verify_ctx(EVP_MD_CTX_new(), EVP_MD_CTX_free);
     int result = EVP_PKEY_encrypt(encrypt_pkey_ctx.get(), encrypted_data.data(), &encrypted_data_length, data.data(),
             data.size());
+    if (result == -2)
```
Comment (mhabrat) [inline reference/src/client/test/sa_engine_pkey_decrypt.cpp:94]: Recommend using defined enums instead of magic numbers for code readability.

**Review by mhabrat — CHANGES_REQUESTED** [review]

File: reference/src/client/src/sa_engine_pkey.c (hunk 1 above)
Comment (ericeberry) [reply reference/src/client/src/sa_engine_pkey.c:941]: Done

File: reference/src/client/test/sa_engine_pkey_decrypt.cpp (hunk 2 above)
Comment (ericeberry) [reply reference/src/client/test/sa_engine_pkey_decrypt.cpp:94]: Done

**Review by mhabrat — APPROVED** [review]

**Pull request merged**